bgpd/bgp_route.c

```diff
@@ -4721,6 +4721,20 @@ void bgp_update(struct peer *peer, const struct prefix *p, uint32_t addpath_id,
 		    && (bgp->inst_type == BGP_INSTANCE_TYPE_DEFAULT)) {
 			leak_success = vpn_leak_to_vrf_update(bgp, pi, prd);
 		}
+		if (SAFI_UNICAST == safi &&
```

Comment:
For each update we receive, we'll check each nlri to see if leaking "this" nlri is configured, correct? Any thoughts on possible performance hits here? Is there any way to test the impact, or should this code be skipped if import is not configured?

Reply:
I think that there is no need for additional checks in this place for the following reasons:
For the list "bgp->vpn_policy[afi].redistribute_export_vrf" , if the configuration related to redistribute vrf is not set, i.e. the current vrf_bgp is not referenced by any other vrf, then this list will be empty, and there will be no further checks for each route.
The list currently stores the other vrf that introduced the current bgp. For example, if vrf A redistributes vrf B, then the bgp->vpn_policy[afi].redistribute_export_vrf of vrf B will store the name of vrf A.